app/graphql/directives.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable class-methods-use-this */
+const { gql } = require('apollo-server'),
+  { SchemaDirectiveVisitor } = require('graphql-tools'),
+  { defaultFieldResolver } = require('graphql'),
+  validateSchema = require('../utils/schemaValidator'),
+  logger = require('../logger');
+
+class ValidateSchemaDirective extends SchemaDirectiveVisitor {
```

Comment:
interesting... im not sure we should be using this kind of implementations, can be a little dark if they grow, but it's nice what you've done for this case